pkg/auth/providers/activedirectory/activedirectory_client.go

```diff
@@ -163,18 +210,18 @@ func (p *adProvider) getPrincipalsFromSearchResult(result *ldapv2.SearchResult,
 				}
 				groupList = append(groupList, grp)
 			}
-			return userPrincipal, groupList, nil
+			return groupList, nil
 		}
-		return userPrincipal, groupPrincipals, err
+		return groupPrincipals, err
 	}
 
-	result, err = lConn.SearchWithPaging(search, 1000)
+	result, err := lConn.SearchWithPaging(search, 1000)
```

Comment:
just curious, did we decide to use the page limit to be 1000 for nested groups?

Reply:
We switch to SearchWithPaging for this issue, `https://github.com/rancher/rancher/issues/13421`, where searching for a group by name could return a large number of groups. And the nestedQuery can also return a large number of groups.

Reply:
I see, thank you!